blockchain/blockdao.go

```diff
@@ -528,6 +538,7 @@ func (dao *blockDAO) putBlock(blk *block.Block) error {
 	if blk.Height() > topHeight {
 		batch.Put(blockNS, topHeightKey, height, "failed to put top height")
 	}
+	batch.Put(blockNS, topHashKey, hash[:], "failed to put top hash")
```

Comment:
Shall we still put it inside `if blk.Height() > topHeight {` condition block, even though we only append block in sequence currently?

Reply:
ok